iceoryx_hoofs/memory/include/iox/bump_allocator.hpp

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2019 by Robert Bosch GmbH. All rights reserved.
+// Copyright (c) 2021 - 2023 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+#ifndef IOX_HOOFS_MEMORY_BUMP_ALLOCATOR_HPP
+#define IOX_HOOFS_MEMORY_BUMP_ALLOCATOR_HPP
+
+#include "iceoryx_hoofs/cxx/expected.hpp"
+#include "iceoryx_hoofs/iceoryx_hoofs_types.hpp"
+
+#include <cstdint>
+
+namespace iox
+{
+
+enum class BumpAllocatorError
+{
+    OUT_OF_MEMORY,
+    REQUESTED_ZERO_SIZED_MEMORY
```

Comment:
See my comment about misuse cases vs. legitimate errors. No change required now but fundamentally there should be a distinction.